samples/dotnet/kernel-syntax-examples/Example29_CustomPlanner.cs

```diff
@@ -0,0 +1,384 @@
+    // DependsOn: TimeSkill named "time"
+    // DependsOn: BingSkill named "bing"
+    private static IDictionary<string, ISKFunction> LoadQASkill(IKernel kernel)
+    {
+        string folder = RepoFiles.SampleSkillsPath();
+        kernel.ImportSkill(new TimeSkill(), "time");
+#pragma warning disable CA2000 // Dispose objects before losing scope
+        var bing = new WebSearchEngineSkill(new BingConnector(Env.Var("BING_API_KEY")));
+#pragma warning restore CA2000 // Dispose objects before losing scope
+        var search = kernel.ImportSkill(bing, "bing");
+
+        return kernel.ImportSemanticSkillFromDirectory(folder, "QASkill");
+    }
+
+    private static IKernel InitializeKernel()
+    {
+        return new KernelBuilder()
+            .WithLogger(ConsoleLogger.Log)
+            .Configure(
+                config =>
+                {
+                    config.AddAzureTextCompletionService(
+                        Env.Var("AZURE_OPENAI_SERVICE_ID"),
+                        Env.Var("AZURE_OPENAI_DEPLOYMENT_NAME"),
+                        Env.Var("AZURE_OPENAI_ENDPOINT"),
+                        Env.Var("AZURE_OPENAI_KEY"));
+
+                    config.AddAzureTextEmbeddingGenerationService(
+                        Env.Var("AZURE_OPENAI_EMBEDDINGS_SERVICE_ID"),
+                        Env.Var("AZURE_OPENAI_EMBEDDINGS_DEPLOYMENT_NAME"),
+                        Env.Var("AZURE_OPENAI_EMBEDDINGS_ENDPOINT"),
+                        Env.Var("AZURE_OPENAI_EMBEDDINGS_KEY"));
+                })
+            .WithMemoryStorage(new VolatileMemoryStore())
+            .Build();
+    }
+}
+
+// Example Skill that can process XML Markup created by ContextQuery
+public class MarkupSkill
+{
+    [SKFunction("Run Markup")]
+    [SKFunctionName("RunMarkup")]
+    public async Task<SKContext> RunMarkupAsync(SKContext context)
+    {
+        var docString = context.Variables.Input;
+        var plan = docString.FromMarkup("Run a piece of xml markup", context);
+
+        Console.WriteLine("Markup plan:");
+        Console.WriteLine(PlanToString(plan));
+        Console.WriteLine();
+
+        var result = await plan.InvokeAsync();
+        context.Variables.Update(result.Result);
+        return context;
+    }
+
+    private static string PlanToString(Plan originalPlan, string indent = " ")
+    {
+        var sb = new StringBuilder($"{indent}Goal: {originalPlan.Description}\n\n{indent}Steps:\n");
+
+        foreach (var step in originalPlan.Steps)
+        {
+            sb.Append(step.Steps.Count switch
+            {
+                0 =>
+                    $"{indent}{indent}- {string.Join(".", step.SkillName, step.Name)} {string.Join(" ", step.NamedParameters.Select(p => $"{p.Key}='{p.Value}'"))}{(step.NamedOutputs.Where(s => s.Key.ToUpper(System.Globalization.CultureInfo.CurrentCulture) != "INPUT").Select(p => $"{p.Key}").FirstOrDefault() is var namedOutputs ? $" => {namedOutputs}" : "")}",
```

Comment:
Can you please just break this out into a private helper function, or a mutli-line body? Break the magic up into variables and then fortmat the string, or use StringBuilder.